codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -265,18 +271,21 @@ public Http2Stream resetSent() {
         }
 
         @Override
-        public final Object setProperty(Object key, Object value) {
-            return data.put(key, value);
+        public final <K, V> V setProperty(PropertyKey<K, V> key, V value) {
+            checkNotNull(key, "key");
+            return properties.add((DefaultPropertyKey<K, V>) key, value);
```

Comment:
@Scottmitch 
Oh I see what you mean, so it wouldn't be flexible. Sounds fine with me. I think in that case we should only do it for the codec, to hide all this complexity from the user.